frontend/scripts/tests/puppeteer/profile-actor.test.js

```diff
@@ -0,0 +1,274 @@
+/* eslint-disable no-console */
+import puppeteer from 'puppeteer';
+
+import mocks from '../mocks';
+
+let page;
+let browser;
+const baseUrl = 'http://0.0.0.0:8081';
+
+const openBrowser = visible =>
+  visible
+    ? {
+        headless: false,
+        slowMo: 80,
+        args: [`--window-size=1920,1080`]
+      }
+    : { args: ['--no-sandbox'] };
+
+beforeAll(async () => {
+  browser = await puppeteer.launch(openBrowser(false));
+  page = await browser.newPage();
+  await page.setRequestInterception(true);
+  page.on('request', interceptedRequest => {
+    const url = interceptedRequest
+      .url()
+      .replace('https:', '')
+      .replace('http:', '');
+
+    if (url in mocks) {
+      console.warn(`URL (${url}) intercepted and response mocked`);
+      setTimeout(
+        () =>
+          interceptedRequest.respond({
+            status: 200,
+            contentType: 'application/json',
+            headers: {
+              'Access-Control-Allow-Origin': '*'
+            },
+            body: JSON.stringify(mocks[url])
+          }),
+        300
+      );
+    } else {
+      interceptedRequest.continue();
+    }
+  });
+});
+
+afterAll(() => {
+  browser.close();
+});
+
+describe('Profile Root search', () => {
+  test(
+    'search for bunge and click importer result',
+    async () => {
+      const nodeName = 'bunge';
+      const nodeType = 'importer';
+      const profileType = 'actor';
+
+      await page.goto(`${baseUrl}/profiles`);
+      await page.waitForSelector('[data-test=profile-search]');
+      await page.click('[data-test=search-input-desktop]');
+      await page.type('[data-test=search-input-desktop]', nodeName);
+      await page.waitForSelector(`[data-test=search-result-${nodeType}-${nodeName}]`);
+      await page.click(`[data-test=search-result-${nodeType}-${nodeName}]`);
+
+      expect(page.url().startsWith(`${baseUrl}/profile-${profileType}`)).toBe(true);
+
+      expect.assertions(1);
+    },
+    30000
+  );
+});
+
+describe('Profile actor', () => {
+  test(
+    'All 5 widget sections attempt to load',
+    async () => {
+      expect.assertions(1);
+
+      // await page.goto(`${baseUrl}/profile-actor?lang=en&nodeId=441&contextId=1&year=2015`);
+      await page.waitForSelector('[data-test=loading-section]');
+      const loadingSections = await page.$$('[data-test=loading-section]');
```

Comment:
all the const you're creating here I would move to another file. Think of that kind of architecture:
`tests/pages/profile.js` where you put all the selectors
`tests/e2e/profile-actor.js` where you import the selectors from pages and have actual tests here, you could also split the tests for specific behaviours

Reply:
I'm wary of pulling thing up too soon, but I agree it doesn't make sense repeating the selector. Perhaps just hoisting this to the top level would do the trick. 👍